hbase-server/src/main/java/org/apache/hadoop/hbase/util/JSONBean.java

```diff
@@ -334,7 +334,7 @@ private static void writeObject(final JsonWriter writer, final Object value) thr
       } else if(value instanceof Number) {
         Number n = (Number)value;
         double doubleValue = n.doubleValue();
-        if (Double.isNaN(doubleValue) || Double.isInfinite(doubleValue)) {
+        if (!Double.isNaN(doubleValue) && !Double.isInfinite(doubleValue)) {
```

Comment:
Make the check consistent with other branches using Double.isFinite()? That seems simple.

Also, it looks like there is some test gap here? Mind adding a quick test?

Reply:
@bharathv my understanding is that branch-1 is required to have java7 compatibility, and I guess it only exists in java8: https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html#isFinite-double-

I agree there is missing test coverage somewhere. I am unfamiliar with the code, so it may take a while to find where to add.

Reply:
Oh sorry, didn't realize isFinite() is Java 8 and above.

> I am unfamiliar with the code, so it may take a while to find where to add.

A simple unit test, create a json using this JSONBean utilities and validate that it parses as expected?

Reply:
Makes sense... this regression is limited to branch-1, but the unit test should go into trunk... should I create a separate JIRA for that?

Reply:
I think its fine to do it under the same jira since they are all related.